### ISSUE TITLE
add --password to `supabase migration fetch`

### ISSUE THREAD

**Issue by BinaryShrub** [supabase/cli#2993]
Almost all other cli commands will accept `-p` but this one won't :(

Hopefully a quick fix for someone who knows what they are doing?

**Comment by sweatybridge** [comment]
I agree we should make the flag more consistent.

A temporary workaround is to set `SUPABASE_DB_PASSWORD=your-pass` env var.

**Comment by Sanket-0510** [comment]
Hey @sweatybridge I would like to work on this

**Comment by sweatybridge** [comment]
Hi @Sanket-0510, thanks for stepping up for this task.

This is how the [password flag](https://github.com/supabase/cli/blob/develop/cmd/db.go#L275) is currently added to commands. It would be great to do the same for other commands which support `--db-url` flag to add the `-p` flag.

Feel free to open a PR and tag me if you run into any issues.

**Comment by Sanket-0510** [comment]
hey but as the db-url already consist of the database password  for eg. - `postgresql://postgres:[YOUR-PASSWORD]@<url>/postgres` why do we provide an option for the password can't we just extract from the URL incase we want to create configuration for the databse. This something popped out in my mind when I saw the db-url structure.

**Comment by sweatybridge** [comment]
Yes, using db-url flag is another alternative that's available today. Most of the time, these 3 flags are always present on the same command

- db-url
- linked
- local

Since linking a project also saves the password locally, so the password flag is not necessary.

Perhaps the only place it's useful is when used together with `--project-ref` or `--project-id` flag.

**Comment by yaten2302** [comment]
Hey @sweatybridge , may I work on this issue?

**Comment by sweatybridge** [comment]
The latest CLI release no longer requires the database password for migration fetch.

```
npx supabase@latest migration fetch
```
